src/Domain/Reflection.php

```diff
@@ -35,24 +36,62 @@ public function __construct($instance)
     /**
      * Sets an private attribute value on the given instance.
      *
-     * @param  string  $attribute
-     * @param array<string>|bool $value
+     * @param string $attribute
+     * @param mixed $value
      *
      * @return \NunoMaduro\PhpInsights\Domain\Reflection
      */
     public function set(string $attribute, $value): Reflection
     {
-        $property = $this->reflectionClass->getProperty($attribute);
-        $property->setAccessible(true);
-        $property->setValue($this->instance, $value);
+        self::setProperty(
+            $this->reflectionClass,
+            $this->instance,
+            $attribute,
+            $value
+        );
 
         return $this;
     }
 
+    /**
+     * @param ReflectionClass $class
+     * @param mixed $instance
+     * @param string $attribute
+     * @param mixed $value
+     *
+     * @throws ReflectionException
+     */
+    private static function setProperty(
+        ReflectionClass $class,
+        $instance,
+        string $attribute,
+        $value
+    ): void
+    {
+        try {
+            $property = $class->getProperty($attribute);
+            $property->setAccessible(true);
+            $property->setValue($instance, $value);
+        } catch (ReflectionException $exception) {
+            $parentClass = $class->getParentClass();
```

Comment:
LOL, nice one. Some dark magic here.

Reply:
Haha yep!
So basically a recursive selector as I had to access private properties of a parent class of the class I selected 🗡